pyaedt/edb_grpc/core/edb_data/layer_data.py

```diff
@@ -0,0 +1,1753 @@
+            self._layer_type = value
+            self.update_layers()
+
+    @property
+    def material_name(self):
+        """Retrieve or update the material name.
+
+        Returns
+        -------
+        str
+            Name of the material.
+        """
+        try:
+            self._material_name = self._cloned_layer.GetMaterial()
+        except:
+            pass
+        return self._material_name
+
+    @material_name.setter
+    def material_name(self, value):
+        self._material_name = value
+        self.update_layers()
+
+    @property
+    def thickness_value(self):
+        """Thickness value.
+
+        Returns
+        -------
+        float
+            Thickness value.
+        """
+        try:
+            self._thickness = self._cloned_layer.GetThicknessValue().ToDouble()
+        except:
+            pass
+        return self._thickness
+
+    @thickness_value.setter
+    def thickness_value(self, value):
+        self._thickness = value
+        self.update_layers()
+
+    @property
+    def filling_material_name(self):
+        """Filling material.
+
+        Returns
+        -------
+        str
+            Name of the filling material if it exists.
+        """
+        if (
+            self._layer_type == self._edb.Cell.LayerType.SignalLayer
+            or self._layer_type == self._edb.Cell.LayerType.ConductingLayer
+        ):
+            try:
+                self._filling_material_name = self._cloned_layer.GetFillMaterial()
+            except:
+                pass
+            return self._filling_material_name
+        return ""
+
+    @filling_material_name.setter
+    def filling_material_name(self, value):
+        if (
+            self._layer_type == self._edb.Cell.LayerType.SignalLayer
+            or self._layer_type == self._edb.Cell.LayerType.ConductingLayer
+        ):
+            self._filling_material_name = value
+            self.update_layers()
+
+    @property
+    def negative_layer(self):
+        """Negative layer.
+
+        Returns
+        -------
+        bool
+            ``True`` when negative, ``False`` otherwise..
+        """
+        if (
+            self._layer_type == self._edb.Cell.LayerType.SignalLayer
+            or self._layer_type == self._edb.Cell.LayerType.ConductingLayer
+        ):
+            try:
+                self._negative_layer = self._layer.GetNegative()
+            except:
+                pass
+        return self._negative_layer
+
+    @negative_layer.setter
+    def negative_layer(self, value):
+        if (
+            self._layer_type == self._edb.Cell.LayerType.SignalLayer
+            or self._layer_type == self._edb.Cell.LayerType.ConductingLayer
+        ):
+            self._negative_layer = value
+            self.update_layers()
+
+    @property
+    def roughness_enabled(self):
+        """Roughness enabled.
+
+        Returns
+        -------
+        bool
+            ``True`` if the layer has roughness, ``False`` otherwise.
```

Comment:
```suggestion
        """Whether roughness is enabled on the layer.

        Returns
        -------
        bool
            ``True`` if roughness is enabled, ``False`` otherwise.
```